frontend/src/index.tsx

```diff
@@ -3,13 +3,19 @@ import ReactDOM from "react-dom";
 import "./index.css";
 import App from "./App";
 import reportWebVitals from "./reportWebVitals";
-import ContextProviders from "./providers/ContextProviders";
+import { PunchlinesContextProvider } from "./contexts/punchlines";
+import { RoundContextProvider } from "./contexts/round";
+import { PlayersContextProvider } from "./contexts/players";
 
 ReactDOM.render(
   <React.StrictMode>
-    <ContextProviders>
-      <App />
-    </ContextProviders>
+    <PlayersContextProvider>
+      <PunchlinesContextProvider>
+        <RoundContextProvider>
+          <App />
+        </RoundContextProvider>
+      </PunchlinesContextProvider>
+    </PlayersContextProvider>
```

Comment:
This nesting looks a little silly lol

Reply:
Yeah - if we switch to [Redux](https://redux.js.org/tutorials/quick-start) it'll reduce down to one 👀